### PR TITLE
feature-toggle-service

### LINKED ISSUE #9: Feature config and toggle service

### PR DESCRIPTION
**Pull request by yeshamavani**
## Description

A feature toggle service using togglz spring library.
1) provides three strategies - System, User, UserTenanat level
2) A method level decorator is available that calls an interceptor to check if the feature is allowed to the user or not.
3) Also  CRUD API are exposed for feature and feature-toggle models. 

Fixes #9 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: sandbox/feature-toggle-example/src/main/java/com/sourcefuse/jarc/sandbox/featuretoggleexample/handlers/HelloHandler.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.sourcefuse.jarc.sandbox.featuretoggleexample.handlers;
+
+import com.sourcefuse.jarc.services.featuretoggleservice.annotation.FeatureHandlers;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component("HelloHandler")
```
Comment (harshadk-sourcefuse) [inline sandbox/feature-toggle-example/src/main/java/com/sourcefuse/jarc/sandbox/featuretoggleexample/handlers/HelloHandler.java]: can we change name of this handler to TenantHandler or an meaningful name

**Review by harshadk-sourcefuse — CHANGES_REQUESTED** [review]

File: sandbox/feature-toggle-example/src/main/java/com/sourcefuse/jarc/sandbox/featuretoggleexample/handlers/HelloHandler.java (hunk 1 above)
Comment (yeshamavani) [reply sandbox/feature-toggle-example/src/main/java/com/sourcefuse/jarc/sandbox/featuretoggleexample/handlers/HelloHandler.java]: Removed were not being used

File: sandbox/feature-toggle-example/src/main/java/com/sourcefuse/jarc/sandbox/featuretoggleexample/FeatureToggleExampleApplication.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
+
+@SpringBootApplication
+@ComponentScan(
+  {
+    "com.sourcefuse.jarc.core",
+    "com.sourcefuse.jarc.services.authservice",
```
Comment (mlakdawala-sf) [inline sandbox/feature-toggle-example/src/main/java/com/sourcefuse/jarc/sandbox/featuretoggleexample/FeatureToggleExampleApplication.java]: you might not need auth-service for this, we can use auth package

File: services/feature-toggle-service/src/main/java/com/sourcefuse/jarc/services/featuretoggleservice/FeatureToggleServiceApplication.java [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 
 @SpringBootApplication
+@ComponentScan(
+  {
+    "com.sourcefuse.jarc.services.featuretoggleservice",
+    "com.sourcefuse.jarc.core",
+    "com.sourcefuse.jarc.services.authservice"
```
Comment (mlakdawala-sf) [inline services/feature-toggle-service/src/main/java/com/sourcefuse/jarc/services/featuretoggleservice/FeatureToggleServiceApplication.java]: you might not need auth-service for this, we can use auth package

**Review by mlakdawala-sf — CHANGES_REQUESTED** [review]

**Review by harshadk-sourcefuse — APPROVED** [review]

File: services/feature-toggle-service/src/main/java/com/sourcefuse/jarc/services/featuretoggleservice/FeatureToggleServiceApplication.java (hunk 3 above)
Comment (yeshamavani) [reply services/feature-toggle-service/src/main/java/com/sourcefuse/jarc/services/featuretoggleservice/FeatureToggleServiceApplication.java]: done

File: sandbox/feature-toggle-example/src/main/java/com/sourcefuse/jarc/sandbox/featuretoggleexample/FeatureToggleExampleApplication.java (hunk 2 above)
Comment (yeshamavani) [reply sandbox/feature-toggle-example/src/main/java/com/sourcefuse/jarc/sandbox/featuretoggleexample/FeatureToggleExampleApplication.java]: done

**Review by mlakdawala-sf — APPROVED** [review]

**Pull request merged**